jumpcloud_bootstrap_template.sh

```diff
@@ -1196,21 +1598,153 @@ if [[ -f $DEP_N_GATE_DONE ]]; then
+        cmdRequest=$(
+            curl -s -o $DEP_N_HTTP_RESPONSE -w "%{http_code}" \
+            -X POST \
+            -H 'content-type: application/json' \
+            -H "x-api-key: $APIKEY" \
+            -d '{"_id":"'"$commandID"'","systemIds":["'"$systemID"'"]}' \
+            --url https://console.jumpcloud.com/api/runCommand
+        )
+        responseContent=$(cat $DEP_N_HTTP_RESPONSE)
+        if [[ $cmdRequest != "200" ]]; then
+            # handle error
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): [error] HTTP Response does not indicate success" >>"$DEP_N_DEBUG"
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): HTTP Response: $responseContent" >>"$DEP_N_DEBUG"
+        else
+            regexResponseWorkflowID='\"workflowInstanceId\":\"([a-zA-Z0-9]{24})\"'
+            if [[ $responseContent =~ $regexResponseWorkflowID ]]; then
+                responseWorkflowID="${BASH_REMATCH[1]}"
+            fi
+            echo "$(date "+%Y-%m-%d %H:%M:%S"): Command ran with workflowID $responseWorkflowID" >>"$DEP_N_DEBUG"
+        fi
+        ## End Run Command
+        ## Begin wait for command results
+        totalCount='0'
+        ResultTimeoutCounter='0'
+        while [[ $totalCount == '0' ]]; do
+            # now get the command result
+            cmdResultRequest=$(
+                curl -s -o $DEP_N_HTTP_RESPONSE -w "%{http_code}" \
+                -X POST \
+                -H 'content-type: application/json' \
+                -H "x-api-key: $APIKEY" \
+                -d '{"filter": {"and": [ "'"workflowId:eq:$commandID"'", "'"workflowInstanceId:eq:$responseWorkflowID"'"]}}' \
+                --url https://console.jumpcloud.com/api/search/commandresults
+            )
+            commandResponseContent=$(cat $DEP_N_HTTP_RESPONSE)
+            if [[ $cmdResultRequest != "200" ]]; then
+                # handle error
+                echo "$(date "+%Y-%m-%d %H:%M:%S"): [error] HTTP Response does not indicate success" >>"$DEP_N_DEBUG"
+            else
+                ## Update Exit Code Variable
+                regexTotalCount='\"totalCount\":([0-9]{1})'
+                if [[ $commandResponseContent =~ $regexTotalCount ]]; then
+                    totalCount="${BASH_REMATCH[1]}"
+                fi
+                ## Log command result count (should be 1 after its run):
+                echo "$(date "+%Y-%m-%d %H:%M:%S"): command results found: $totalCount" >>"$DEP_N_DEBUG"
+            fi
+
+            # If no command results are found, wait
+            if [[ $totalCount == '0' ]]; then
+                echo "$(date "+%Y-%m-%d %H:%M:%S"): Expected command result not found, waiting 10s..." >>"$DEP_N_DEBUG"
+                ResultTimeoutCounter=$((${ResultTimeoutCounter} + 1))
+                sleep 10
+            fi
+            if [[ $ResultTimeoutCounter -eq 24 ]]; then
+                echo "$(date "+%Y-%m-%d %H:%M:%S"): Error during JumpCloud Command Result Lookup" >>"$DEP_N_DEBUG"
+                break
```

Comment:
At this stage there's not much we can do if the enrollment users are not removed w/ the JumpCloud Command, this would be reflected in the log.